src/App.js

```diff
@@ -1,23 +1,122 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+import contactsData from "./contacts.json";
+import { useState } from "react";
 
 function App() {
+  const [contacts, setContacts] = useState(contactsData.slice(0, 5));
+
+  function addRandom() {
+    const randomContact =
+      contactsData[
+        Math.floor(Math.random() * (contactsData.length - 1 - 5) + 5)
+      ];
+    setContacts([...contacts, randomContact]);
+  }
+
+  function sortByName() {
+    const sortedContacts = [...contacts].sort((a, b) => {
+      if (a.name > b.name) {
+        return 1;
+      } else if (a.name < b.name) {
+        return -1;
+      } else {
+        return 0;
+      }
+    });
+    setContacts(sortedContacts);
+  }
+
+  function sortByPopularity() {
+    const sortedContacts = [...contacts].sort((a, b) => {
+      if (a.popularity > b.popularity) {
+        return -1;
+      } else if (a.popularity < b.popularity) {
+        return +1;
+      } else {
+        return 0;
+      }
+    });
+    setContacts(sortedContacts);
+  }
+
+  function deleteContact(contactId) {
+    const filteredContacts = contacts.filter(
+      (contact) => contact.id !== contactId
+    );
+    setContacts(filteredContacts);
+  }
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <h1>
+        IronContacts{" "}
+        <img src="/movie-clapper-open.png" alt="movie-clapper-icon" />
+      </h1>
+      <div>
+        <button className="btn primary" onClick={addRandom}>
+          Add Random Contact
+        </button>
+        <button className="btn primary" onClick={sortByName}>
+          Sort by name
+        </button>
+        <button className="btn primary" onClick={sortByPopularity}>
+          Sort by popularity
+        </button>
+      </div>
+      {contacts.map((oneContact) => {
+        return (
+          <div>
```

Comment:
Remember when we use .map its oportant to add the key prop so react can difference which node you are refering
```suggestion
          <div key={oneContact.id}>
```